pstoremem/addr_book.go

```diff
@@ -98,6 +110,7 @@ func (mab *memoryAddrBook) gc() {
 	now := time.Now()
 	for _, s := range mab.segments {
 		s.Lock()
+		var collectedPeers []peer.ID
```

Comment:
Could use a single slice for all segments, resetting to `[:0]` on each iteration. That would produce a smaller memory footprint.